### DISCUSSION TITLE
Any docker command even docker pull will response a error when use new rancher desktop version: http: server gave HTTP response to HTTPS client

### DISCUSSION THREAD

**Discussion by aib628** [rancher-sandbox/rancher-desktop#7520] [Q&A] [answered]
Any docker command rasponse 'http: server gave HTTP response to HTTPS client' after upgrade my rancher desktop version from 1.10.0 to 1.16.0

Error response from daemon: Head "https://registry-1.docker.io/v2/xxx/hadoop-historyserver/manifests/3.3.6": Get "https://auth.docker.io/token?scope=repository%3Axxx%2Fhadoop-historyserver%3Apull&service=registry.docker.io": http: server gave HTTP response to HTTPS client
![image](https://github.com/user-attachments/assets/2f50b6ad-a490-4fef-9ac8-359279f8e553)

**Comment by mook-as** [comment]
Hmm, do you have any proxies configured? What operating system is Rancher Desktop running on? Would you be able to [enable debug logging](https://docs.rancherdesktop.io/ui/troubleshooting/#enable-debug-mode) and [attach the logs](https://docs.rancherdesktop.io/ui/troubleshooting/#show-logs)?

**Reply by aib628** [reply]
> Hmm, do you have any proxies configured? What operating system is Rancher Desktop running on? Would you be able to [enable debug logging](https://docs.rancherdesktop.io/ui/troubleshooting/#enable-debug-mode) and [attach the logs](https://docs.rancherdesktop.io/ui/troubleshooting/#show-logs)?

Operating System: MacOS 14.5 aarch
Kubernetes Version: 1.60.0

After enable debug mode, retry login, there is no log record
```
Failed to start web-based login - falling back to command line login...

Log in with your Docker ID or email address to push and pull images from Docker Hub. If you don't have a Docker ID, head over to https://hub.docker.com/ to create one.
You can log in with your password or a Personal Access Token (PAT). Using a limited-scope PAT grants better security and is required for organizations using SSO. Learn more at https://docs.docker.com/go/access-tokens/

Username: xxx
Password:
Error response from daemon: Get "https://registry-1.docker.io/v2/": Get "https://auth.docker.io/token?account=xxx&client_id=docker&offline_token=true&service=registry.docker.io": http: server gave HTTP response to HTTPS client
(base) vmaster@/Users/vmaster/Library/Logs/rancher-desktop $ ll
total 408
-rw-------@ 1 vmaster  staff     0B  9 23 10:43 background.log
-rw-------@ 1 vmaster  staff     0B  9 23 10:43 commandLine.log
-rw-------@ 1 vmaster  staff     0B  9 23 10:43 dashboardServer.log
-rw-------@ 1 vmaster  staff     0B  9 23 10:43 deploymentProfile.log
-rw-------@ 1 vmaster  staff   147K  9 23 11:19 diagnostics.log
-rw-------@ 1 vmaster  staff     0B  9 23 10:43 extensions.log
-rw-------@ 1 vmaster  staff     0B  9 23 10:43 images.log
-rw-------@ 1 vmaster  staff     0B  9 23 10:43 integrations.log
-rw-r--r--@ 1 vmaster  staff   3.0K  9 23 11:18 k3s.log
-rw-------@ 1 vmaster  staff     0B  9 23 10:43 k8s.log
-rw-------@ 1 vmaster  staff     0B  9 23 10:43 kube.log
lrwxr-xr-x  1 vmaster  staff    62B  9 20 12:04 lima.ha.stderr.log -> ../../Application Support/rancher-desktop/lima/0/ha.stderr.log
lrwxr-xr-x  1 vmaster  staff    62B  9 20 12:04 lima.ha.stdout.log -> ../../Application Support/rancher-desktop/lima/0/ha.stdout.log
-rw-------@ 1 vmaster  staff     0B  9 23 10:43 lima.log
lrwxr-xr-x  1 vmaster  staff    59B  9 20 12:04 lima.serial.log -> ../../Application Support/rancher-desktop/lima/0/serial.log
lrwxr-xr-x@ 1 vmaster  staff    60B  9 20 12:14 lima.serialp.log -> ../../Application Support/rancher-desktop/lima/0/serialp.log
lrwxr-xr-x@ 1 vmaster  staff    60B  9 20 12:14 lima.serialv.log -> ../../Application Support/rancher-desktop/lima/0/serialv.log
-rw-------@ 1 vmaster  staff     0B  9 23 10:43 moby.log
-rw-------@ 1 vmaster  staff     0B  9 23 10:43 mock.log
-rw-------@ 1 vmaster  staff     0B  9 23 10:43 nerdctl.log
-rw-------@ 1 vmaster  staff     0B  9 23 10:43 networking.log
-rw-------@ 1 vmaster  staff     0B  9 23 10:43 path-management.log
-rw-------@ 1 vmaster  staff     0B  9 23 10:43 protocol-handler.log
-rw-------@ 1 vmaster  staff     0B  9 23 10:43 server.log
-rw-------@ 1 vmaster  staff     0B  9 23 10:43 settings.log
-rw-------@ 1 vmaster  staff     0B  9 23 10:43 shortcuts.log
-rw-------@ 1 vmaster  staff     0B  9 23 10:43 snapshots.log
-rw-------@ 1 vmaster  staff     0B  9 23 10:43 steve.log
-rw-------@ 1 vmaster  staff     0B  9 23 10:43 update.log
-rw-------@ 1 vmaster  staff     0B  9 23 10:43 window_browser.log
-rw-------@ 1 vmaster  staff     0B  9 23 10:43 wsl-version.log
-rw-------@ 1 vmaster  staff     0B  9 23 10:43 wsl.log
-rw-r--r--  1 vmaster  staff   459B  9 23 10:44 {}
```

There is no proxy configured，and i have tried:
1、connecting to the company intranet and switch to mobile hotspot.
2、Reset Kubernetes
3、Factory Reset
4、Reboot rancher desktop
5、Reinstall rancher desktop

**Reply by mook-as** [reply, marked as answer]
Can you manually visit `https://auth.docker.io/` in a browser? _server gave HTTP response to HTTPS client_ sounds like there's some sort of network interception going on and it's trying to get you to do something to allow access.

**Reply by aib628** [reply]
> Can you manually visit `https://auth.docker.io/` in a browser? _server gave HTTP response to HTTPS client_ sounds like there's some sort of network interception going on and it's trying to get you to do something to allow access.

The browser is also unable to access it because of there is no speed up service for some reason, thank you very much !
By the way, where should i to config my rancher desktop such as /etc/docker/daemon.json of docker desktop? eg: `registry-mirrors` and `insecure-registries`

**Reply by mook-as** [reply]
I don't understand what a _speed up service_ is, but as long as you've got that resolved I guess that's good.

You can run `rdctl shell` to be inside the VM to edit `/etc/docker/daemon.json`. Tools inside the VM are kind of limited, though.

**Reply by aib628** [reply]
> I don't understand what a _speed up service_ is, but as long as you've got that resolved I guess that's good.
> 
> You can run `rdctl shell` to be inside the VM to edit `/etc/docker/daemon.json`. Tools inside the VM are kind of limited, though.

Thanks for your reply. the speed up service is a network proxy of my company, so i can't visit `https://auth.docker.io/` in a browser, and also can't login to docker even after config `"insecure-registries": ["http://registry-1.docker.io"] or "insecure-registries": ["registry-1.docker.io"]`
